Add SSE to SimpleTable

oharaandrew314 opened the pull request:
Closes #376 

Adding optional [SSE](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-ssespecification.html) to `AWS::Serverless::SimpleTable`.  See #376 for more info.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

oharaandrew314 commented on samtranslator/model/sam_resources.py:
```diff
@@ -1,6 +1,4 @@
 """ SAM macro definitions """
-from copy import deepcopy
```
I fixed a bunch of flake errors as I was passing through here.

oharaandrew314 commented on samtranslator/model/sam_resources.py:
```diff
@@ -510,7 +512,8 @@ class SamSimpleTable(SamResourceMacro):
         'TableName': PropertyType(False, one_of(is_str(), is_type(dict))),
-        'Tags': PropertyType(False, is_type(dict))
+        'Tags': PropertyType(False, is_type(dict)),
+        'SSESpecification': PropertyType(False, is_type(dict))
```
This is a functional change

oharaandrew314 commented on samtranslator/model/sam_resources.py:
```diff
@@ -549,6 +551,9 @@ def _construct_dynamodb_table(self):
 
         dynamodb_table.ProvisionedThroughput = provisioned_throughput
 
+        if self.SSESpecification:
```
This is a functional change

oharaandrew314 commented on samtranslator/model/dynamodb.py:
```diff
@@ -13,8 +13,9 @@ class DynamoDBTable(Resource):
-            'Tags': PropertyType(False, list_of(is_type(dict)))
-        }
+            'Tags': PropertyType(False, list_of(is_type(dict))),
+            'SSESpecification': PropertyType(False, list_of(is_type(dict)))
```
Whoops.  I wonder how this ever worked.

jfuss replied:
Ya. I don't think we have good testing around these, mainly because they are the CloudFormation Resources as objects. :(

The line is required but they second param to `PropertyType` should be `is_type(dict)`.

Background, there python magic happening to produce the actually classes. All the vanilla CloudFormation Resources, inherit from `Resource` and in the `__init__` there is this [magical code that creates the properties in the class](https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/model/__init__.py#L66). I find this confusing and took me a while to understand but it is what it is.

oharaandrew314 commented:
@jfuss, I believe I've addressed all your concerns.

SimpleTable did not have any supported globals, so I've added SSESpecification as the only one.

jfuss commented:
> SimpleTable did not have any supported globals, so I've added SSESpecification as the only one.

Thanks for adding that. 

I am taking a look through the PR again now :)

jfuss commented:
@oharaandrew314 I will do my best to pull this in today. I need to figure out how to we will communicate features that are not yet deployed but are updated here. Staging it here until I have clarity on that.

oharaandrew314 commented:
Yay!  So, is there now a process for handling merged versus releases features?

brettstack commented:
Yes. We'll cut a release branch and not merge to master until the feature is available.
